gammapy/modeling/selection.py

```diff
@@ -61,15 +62,46 @@ def ts_threshold(self):
         This assumes that the TS follows a chi squared distribution
         with a number of degree of freedom equal to `n_free_parameters`.
         """
-        return sigma_to_ts(self.n_sigma, self.n_free_parameters)
+        return np.sign(self.n_sigma) * sigma_to_ts(self.n_sigma, self.n_free_parameters)
```

Comment:
I introduced that here such that setting  n_sigma=-np.inf gives ts=-np.inf so the selection is for sure disabled.
I use this rather than n_sigma=0 because depending on the fit convergence ts can be <0 (slightly unless if the fit fails).